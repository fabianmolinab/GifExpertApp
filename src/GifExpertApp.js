import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaulCategories = [] }) => {

  const [categorias, setCategories] = useState(defaulCategories);

  return (
    <>
      <h2>Â¡Puedes buscar cualquier Gif!</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {
          categorias.map(category =>
            <GifGrid
              category={category}
              key={category}
            />
          )
        }
      </ol>
    </>
  )
};