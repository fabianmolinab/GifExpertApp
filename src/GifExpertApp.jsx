import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
  
  const [categorias, setCategories] = useState([]);

  return ( 
    <>
      <h2>GifExpertApp</h2> 
      <AddCategory setCategories={ setCategories }/>
      <hr/>

      <ol>
        {
          categorias.map( category => 
            <GifGrid 
              category={ category }
              key={ category }
            />
          )
        }
      </ol>
    </>
  )
};