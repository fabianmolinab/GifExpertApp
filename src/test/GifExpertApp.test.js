import React from 'react';

import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe(' Pruebas de GifExpertApp ', () => {

  test(' Debe mostrar correctamente el componente ', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot();
  })

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['Fabian', 'Love'];

    const wrapper = shallow(< GifExpertApp defaulCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);

  })

})