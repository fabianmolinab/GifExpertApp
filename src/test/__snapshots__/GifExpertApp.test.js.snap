// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas de GifExpertApp   Debe mostrar correctamente el componente  1`] = `
<Fragment>
  <h2>
    ¡Puedes buscar cualquier Gif!
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[` Pruebas de GifExpertApp  debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    ¡Puedes buscar cualquier Gif!
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Fabian"
      key="Fabian"
    />
    <GifGrid
      category="Love"
      key="Love"
    />
  </ol>
</Fragment>
`;
